swisscovid-cn-sdk-backend/swisscovid-cn-sdk-backend-ws/src/test/java/ch/ubique/swisscovid/cn/sdk/backend/ws/insertmanager/insertfilters/OverlappingIntervalsFilterTest.java

```diff
@@ -4,58 +4,105 @@
 
 import ch.ubique.swisscovid.cn.sdk.backend.model.UserUploadPayloadOuterClass.UploadVenueInfo;
 import ch.ubique.swisscovid.cn.sdk.backend.ws.insertmanager.OverlappingIntervalsException;
+import ch.ubique.swisscovid.cn.sdk.backend.ws.insertmanager.insertfilters.OverlappingIntervalsFilterTest.IntervalTestVectors.CheckinParams;
+import ch.ubique.swisscovid.cn.sdk.backend.ws.insertmanager.insertfilters.OverlappingIntervalsFilterTest.IntervalTestVectors.TestCase;
+import ch.ubique.swisscovid.cn.sdk.backend.ws.util.CryptoUtil.NoncesAndNotificationKey;
+import com.google.gson.Gson;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStreamReader;
 import java.time.LocalDateTime;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import org.junit.Before;
 import org.junit.Test;
 
 public class OverlappingIntervalsFilterTest extends UploadInsertionFilterTest {
+
+    private IntervalTestVectors intervalTestVectors;
+
+    @Before
+    public void init() throws FileNotFoundException {
+        intervalTestVectors =
+                new Gson()
+                        .fromJson(
+                                new InputStreamReader(
+                                        new FileInputStream(
+                                                "src/test/resources/crowd_notifier_overlapping_intervals.json")),
+                                IntervalTestVectors.class);
+    }
+
     @Override
-    List<UploadVenueInfo> getValidVenueInfo() {
-        final var venueInfoList = new ArrayList<UploadVenueInfo>();
-        LocalDateTime start = LocalDateTime.now();
-        LocalDateTime end = start.plusHours(1);
-        final var venueInfo1 = getVenueInfo(start, end);
-        venueInfoList.add(venueInfo1);
-        start = end.plusMinutes(1);
-        end = start;
-        final var venueInfo2 = getVenueInfo(start, end);
-        venueInfoList.add(venueInfo2);
-        return venueInfoList;
+    List<List<UploadVenueInfo>> getValidVenueInfo() {
```

Comment:
is the test for the valid cases missing here?

Reply:
No. In this case, we only needed to override the test driver for the invalid test cases, as the `OverlappingIntervalsFilter` doesn't just filter out invalid `VenueInfo`s, but also throws an exception. When you run all test cases in `OverlappingIntervalsFilterTest`, you'll see that the valid test cases are also run.